guides/common/modules/proc_rex-pull-based-provider.adoc

```diff
@@ -0,0 +1,31 @@
+[id="REX_pull_based_provider_{context}"]
+= REX pull based provider 
+
+Remote Execution has been around for a while now and has served SSH-based execution very well.
+ 
+Now, {Project} has a ‘pull’ based provider that has the features of REX but with a deployment model similar to katello-agent. 
+
+This provider uses MQTT as its messaging protocol and includes an MQTT client running on each content host.
+
+In order to tune the MQTT server (`mosquitto`) and increase the number of clients (content hosts) connnected to it, you can follow the following procedure on the {Project} server:
+
+.Procedure
+. Enable the feature using: `{foreman-installer} --foreman-proxy-plugin-remote-execution-script-mode pull-mqtt` (this will change the REX transport mode from the default SSH one to MQTT).
+  It's important to mention that one {Project} server ({ProjectServer} or {SmartProxyServer}) can only use one transport mode: SSH or MQTT.
+
+. To increase the default number of clients accepted by the MQTT server, create the following system unit definition file and set its contents as specified in the below given example (this will allow the `mosquitto` service to handle ~5000 clients):
```

Comment:
```suggestion
. To increase the default number of hosts accepted by the MQTT server, create the following system unit definition file and set its contents as specified in the below given example (this will allow the `mosquitto` service to handle ~5000 clients):
```

Reply:
Is this optional? What's the default value? Is 5000 clients a recommendation? Does it depends on the specs/used tuning profile?

Reply:
It is optional, but there might be dragons, depending on your scale. According to systemd docs[1] the default limit is 1024. Based on the reasoning in[1], I wouldn't say it depends on specs/tuning profile.

[1] - https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Process%20Properties

Reply:
Please see https://github.com/theforeman/foreman-documentation/pull/1747#discussion_r1009725326 for more discussion. I'd say that if we raise beyond 5000, we might as well raise it WAY higher. Then if we do want to limit it, we should use `max_connections` in `mosquitto.conf` instead.

Reply:
We chose 5000 because it's what we were able to test, but agree with your reasoning.